optimum/onnxruntime/modeling_ort.py

```diff
@@ -152,68 +136,47 @@ class ORTModel(OptimizedModel):
         - auto_model_class (`Type`, *optional*, defaults to `AutoModel`) -- The "AutoModel" class to represented by the
         current ORTModel class.
 
-    Common attributes:
-        - model (`ort.InferenceSession`) -- The ONNX Runtime InferenceSession that is running the model.
+    Args:
         - config ([`~transformers.PretrainedConfig`] -- The configuration of the model.
+        - session (`~onnxruntime.InferenceSession`) -- The ONNX Runtime InferenceSession that is running the model.
         - use_io_binding (`bool`, *optional*, defaults to `True`) -- Whether to use I/O bindings with **ONNX Runtime
         with the CUDAExecutionProvider**, this can significantly speedup inference depending on the task.
         - model_save_dir (`Path`) -- The directory where the model exported to ONNX is saved.
         By defaults, if the loaded model is local, the directory where the original model will be used. Otherwise, the
         cache directory is used.
-        - providers (`List[str]) -- The list of execution providers available to ONNX Runtime.
     """
 
     model_type = "onnx_model"
     auto_model_class = AutoModel
 
-    @classproperty
-    def export_feature(cls):
-        logger.warning(f"{cls.__name__}.export_feature is deprecated, and will be removed in optimum 2.0.")
-        try:
-            feature = TasksManager.infer_task_from_model(cls.auto_model_class)
-        except ValueError:
-            feature = None
-        return feature
-
-    @classmethod
-    def _auto_model_to_task(cls, auto_model_class):
-        """
-        Get the task corresponding to a class (for example AutoModelForXXX in transformers).
-        """
-        return TasksManager.infer_task_from_model(auto_model_class)
-
-    def shared_attributes_init(
+    def __init__(
         self,
-        model: ort.InferenceSession,
+        config: "PretrainedConfig",
+        session: "InferenceSession",
```

Comment:
not against but breaking change so not sure that's worth it (could result in issues for users who don't use .from_pretrained)

Reply:
I agree, this is the change I'm hesitant about the most 😅
I think we can do it slowly by using *args and warning users from passing positional arguments to the ORTModel class.

Reply:
I pushed a "solution" that (hopefully) doesn't break BC, tell me what you think.
I see that some libs are using ORTModelForCausalLM and instantiating it directly (mostly intel/onnx neural-compressor).
We can keep this until optimum 2.0